plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstance.java

```diff
@@ -98,6 +101,7 @@
     @Inject
     public DockerInstance(DockerConnector docker,
                           @Named("machine.docker.registry") String registry,
+                          @Named("docker.registry.auth.username") @Nullable String username,
```

Comment:
Then it should be inside method that generates repository. It should not be username because it there is no username in repo. There are only parts of repo and some of them could implement some namespace. But it is not mandatory, it can include not 1 part of repo (separated by slashes) and it should not be called username.